sdk/servicebus/service-bus/src/receiver.ts

```diff
@@ -74,12 +74,20 @@ export class Receiver {
     if (!onError || typeof onError !== "function") {
       throw new Error("'onError' is a required parameter and must be of type 'function'.");
     }
-    const sReceiver = StreamingReceiver.create(this._context, {
+    StreamingReceiver.create(this._context, {
       ...options,
       receiveMode: this._receiveMode
-    });
-    this._context.streamingReceiver = sReceiver;
-    return sReceiver.receive(onMessage, onError);
+    })
+      .then(async (sReceiver) => {
+        if (!this.isClosed && sReceiver) {
+          return sReceiver.receive(onMessage, onError);
+        } else {
+          await sReceiver.close();
```

Comment:
We will enter the `else` block when `sReciever` is undefined/null as well. I would suggest an early exit using
```
if (!sReceiver) {
    return;
}
```

Reply:
Technically, this wouldn't happen since we synchronously create the object, but agreed!